src/test/java/dog/pawbook/logic/commands/EditDogCommandTest.java

```diff
@@ -0,0 +1,122 @@
+import static dog.pawbook.testutil.TypicalId.ID_TWO;
+import static java.util.stream.Collectors.toList;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+
+import dog.pawbook.commons.core.Messages;
+import dog.pawbook.logic.commands.EditDogCommand.EditDogDescriptor;
+import dog.pawbook.model.Database;
+import dog.pawbook.model.Model;
+import dog.pawbook.model.ModelManager;
+import dog.pawbook.model.UserPrefs;
+import dog.pawbook.model.managedentity.Entity;
+import dog.pawbook.model.managedentity.IdMatchPredicate;
+import dog.pawbook.model.managedentity.dog.Dog;
+import dog.pawbook.testutil.DogBuilder;
+import dog.pawbook.testutil.EditDogDescriptorBuilder;
+import javafx.util.Pair;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for EditDogCommand.
+ */
+public class EditDogCommandTest {
+
+    private Model model = new ModelManager(getTypicalDatabase(), new UserPrefs());
+
+    @Test
+    public void execute_allFieldsSpecified_success() {
+        Pair<Integer, Entity> firstIdEntity = model.getDatabase().getEntityList().get(1);
+        Dog editedDog = new DogBuilder((Dog) firstIdEntity.getValue()).build();
+        EditDogCommand.EditDogDescriptor descriptor = new EditDogDescriptorBuilder(editedDog).build();
+        EditDogCommand editDogCommand = new EditDogCommand(firstIdEntity.getKey(), descriptor);
+
+        String expectedMessage = String.format(EditDogCommand.MESSAGE_EDIT_DOG_SUCCESS, editedDog);
+
+        Model expectedModel = new ModelManager(new Database(model.getDatabase()), new UserPrefs());
+        expectedModel.setEntity(firstIdEntity.getKey(), editedDog);
+        expectedModel.updateFilteredEntityList(new IdMatchPredicate(firstIdEntity.getKey()));
+        assertCommandSuccess(editDogCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_someFieldsSpecified_success() {
+        Dog toEditDog = (Dog) model.getEntity(ID_TWO);
+
+        DogBuilder dogInList = new DogBuilder(toEditDog);
+        Dog editedDog = dogInList.withName(VALID_NAME_BELL).withBreed(VALID_BREED_BELL)
+                .withTags(VALID_TAG_FRIENDLY).build();
+
+        EditDogCommand.EditDogDescriptor descriptor = new EditDogDescriptorBuilder().withName(VALID_NAME_BELL)
+                .withBreed(VALID_BREED_BELL).withTags(VALID_TAG_FRIENDLY).build();
+        EditDogCommand editEntityCommand = new EditDogCommand(ID_TWO, descriptor);
+
+        String expectedMessage = String.format(EditDogCommand.MESSAGE_EDIT_DOG_SUCCESS, editedDog);
+
+        Model expectedModel = new ModelManager(new Database(model.getDatabase()), new UserPrefs());
+        expectedModel.setEntity(ID_TWO, editedDog);
+        expectedModel.updateFilteredEntityList(new IdMatchPredicate(ID_TWO));
+
+        assertCommandSuccess(editEntityCommand, model, expectedMessage, expectedModel);
+    }
+
```

Comment:
add a test that changes the owner ID, and also one that causes a duplicate, change the other 2 accordingly as well as you deem fit.

Reply:
To avoid any confusion, I meant to **add** an additional test for this, not to rename this test method.

Reply:
Done!